Add a SelectionMode shorthand for Grid.

pleku opened the pull request:
Hides setSelectionModel(...) by making it protected.
Adds shortcuts for Grid to set select all checkbox visible, if SelectionMode.MULTI is used.
Refactores the usage of constructor / extend method for Abstract-, Single- and MultiSelectionModelImpl.

Fixes vaadin/framework8-issues#519

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/83)
<!-- Reviewable:end -->

Commit:
Add a SelectionMode shorthand for Grid.

Hides setSelectionModel(...) by making it protected.
Refactores the usage of constructor / extend method for Abstract-, Single- and MultiSelectionModelImpl.

Fixes vaadin/framework8-issues#519

Change-Id: I48c30886450506639be9ee6e21c45b0c06755c88

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/framework?pullRequest=83) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/framework?pullRequest=83) before we can accept your contribution.<br/>

pleku commented:
Review status: 0 of 17 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[server/src/test/java/com/vaadin/tests/components/grid/GridSingleSelectionModelTest.java, line 126 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzaKKN0TdZae8bzVKh:-KXzaKKO5UdTC9k01iGb:bm23atg) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/test/java/com/vaadin/tests/components/grid/GridSingleSelectionModelTest.java#L126)):*
> ```Java
>         // switch to another selection model to cause event
>         customModel.generatedData.clear();
>         customGrid.setSelectionMode(SelectionMode.MULTI);
> ```

Switches to different selection model, but that doesn't affect the test.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 17 of 17 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java, line 0 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzceQd1OuwL8m7fwfv:-KXzceQd1OuwL8m7fwfw:bpj0d22) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java#L0)):*
But this test is for v7.Grid, not for modern Grid.
Are you sure that it has no sense anymore ?

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 222 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzd-KHCMdV_QXEIwwk:-KXzd-KHCMdV_QXEIwwl:b-qdahti) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/main/java/com/vaadin/ui/Grid.java#L222)):*
> ```Java
>          *             if this selection model has been bound to a grid already
>          */
>         public void extend(Grid<T> grid);
> ```

Weird name for the name. I guess it's because of Extension.
I'm as a developer cannot get from what it means.

I wouldn't expose it at all.....
And `Extension` here as a super interface is also weird for me (it's rather an impl detail).
But this is out of the scope of the change.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3132 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXze-JB7ZdBwmYUXYV5:-KXze-JB7ZdBwmYUXYV6:boed7rz) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/main/java/com/vaadin/ui/Grid.java#L3132)):*
> ```Java
>         }
>         selectionModel = model;
>         selectionModel.extend(this);
> ```

Previously it has been implemented without this somehow.......
Is it possible to avoid public `extend` method ?

---

*[server/src/main/java/com/vaadin/ui/components/grid/NoSelectionModel.java, line 66 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzeZccV-GPaZWumVLf:-KXzeZccV-GPaZWumVLg:b-d0wxhj) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/main/java/com/vaadin/ui/components/grid/NoSelectionModel.java#L66)):*
> ```Java
>     }
> 
> }
> ```

new line

---

*[server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java, line 135 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzf-yHmZVCJ24wnPrd:-KXzf-yHmZVCJ24wnPre:burg3vs) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java#L135)):*
> ```Java
>         CustomMultiSelectionModel customModel = new CustomMultiSelectionModel();
> 
>         Grid<String> customGrid = new Grid<String>() {
> ```

Looks like it's repeating construction int the code.
I would extract this as a private nested class and use this child class in each such place instead.

---

*[server/src/test/java/com/vaadin/tests/components/grid/GridSingleSelectionModelTest.java, line 126 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzaKKN0TdZae8bzVKh:-KXzfQAUq-TrNUD-33PI:bjtzpak) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/test/java/com/vaadin/tests/components/grid/GridSingleSelectionModelTest.java#L126)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

Switches to different selection model, but that doesn't affect the test.
</blockquote></details>

I'm just curios : why ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

pleku commented:
Review status: 13 of 18 files reviewed at latest revision, 6 unresolved discussions.

---

*[compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java, line  at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzceQd1OuwL8m7fwfv:-KXzi5VSaZD2q3iojvKe:b-r77p8s) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java#L)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

But this test is for v7.Grid, not for modern Grid.
Are you sure that it has no sense anymore ?
</blockquote></details>

We have a ticket about migrating all remaining old component V7 component tests to V8. I'm removing this test since the V8 grid has the corresponding tests, and these do not need migrating to that anymore. 

The original reasoning was ,that the V7 grid is tested in the V7 build, and V7 components in V8 legacy package should just match that. We have a ticket about creating just "smoke-tests" for V8 build for compatibility components.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 222 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzd-KHCMdV_QXEIwwk:-KXzine_WK4jIzSEGouJ:bbvprqv) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/main/java/com/vaadin/ui/Grid.java#L222)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

Weird name for the name. I guess it's because of Extension.
I'm as a developer cannot get from what it means.

I wouldn't expose it at all.....
And `Extension` here as a super interface is also weird for me (it's rather an impl detail).
But this is out of the scope of the change.
</blockquote></details>

I added this for clarity on the mess what extensions are.. but I've removed it and tried to take the mess into account in setSelectionModel(...)

The problem is that Extension interface doesn't specify other than remove() and setParent(), whereas the actual API should be about AbstractExtension.extend(...) which is used here.

Not really more I can do in this patch.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3132 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXze-JB7ZdBwmYUXYV5:-KXzv_DF54ZqYsZ4LXNH:b-dy3ouy) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/main/java/com/vaadin/ui/Grid.java#L3132)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

Previously it has been implemented without this somehow.......
Is it possible to avoid public `extend` method ?
</blockquote></details>

I made it so that you can create the selection model without directly setting grid to it on constructor. I think it is better that way and now it is up to this method to make sure that the selection model is attached properly to the grid.

---

*[server/src/main/java/com/vaadin/ui/components/grid/NoSelectionModel.java, line 66 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzeZccV-GPaZWumVLf:-KXzvGqPrdrZCAJjWQKV:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/main/java/com/vaadin/ui/components/grid/NoSelectionModel.java#L66)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

new line
</blockquote></details>

Done.

---

*[server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java, line 135 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzf-yHmZVCJ24wnPrd:-KXzy5lAF3AvrMVnqYV7:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java#L135)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

Looks like it's repeating construction int the code.
I would extract this as a private nested class and use this child class in each such place instead.
</blockquote></details>

Done.

---

*[server/src/test/java/com/vaadin/tests/components/grid/GridSingleSelectionModelTest.java, line 126 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzaKKN0TdZae8bzVKh:-KXzvIKTwEsvhjGukpp7:b-x8hg71) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/test/java/com/vaadin/tests/components/grid/GridSingleSelectionModelTest.java#L126)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

I'm just curios : why ?
</blockquote></details>

Since it tests just changing the selection model into something else than what was before, the visible rows should be updated.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

pleku commented:
Review status: 13 of 18 files reviewed at latest revision, 7 unresolved discussions.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3122 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KY00qDiDTetE79pZEQs:-KY00qDjBAOul8HZJz_T:bqx4lqd) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/main/java/com/vaadin/ui/Grid.java#L3122)):*
> ```Java
> 
>         if (selectionModel instanceof AbstractListingExtension) {
>             ((AbstractListingExtension<T>) selectionModel).extend(this);
> ```

AbstractExtension.extend() will trigger the same addExtension used below. Using AbstractListingExtension because AbstractExtension.extend() is protected...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 3 of 5 files at r2.
Review status: 16 of 18 files reviewed at latest revision, 6 unresolved discussions.

---

*[compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java, line  at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzceQd1OuwL8m7fwfv:-KYCg12e9IyMOPBVYiyY:b-y531db) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java#L)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

We have a ticket about migrating all remaining old component V7 component tests to V8. I'm removing this test since the V8 grid has the corresponding tests, and these do not need migrating to that anymore. 

The original reasoning was ,that the V7 grid is tested in the V7 build, and V7 components in V8 legacy package should just match that. We have a ticket about creating just "smoke-tests" for V8 build for compatibility components.
</blockquote></details>

OK.......
But this is not a UI test (so it's executed). Smoke tests are good but is it really necessary to delete existing tests ? (which are already written)
Well, up to you.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 222 at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzd-KHCMdV_QXEIwwk:-KYCgng891HIcQg46qVS:blid5k8) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/server/src/main/java/com/vaadin/ui/Grid.java#L222)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary><blockquote>

I added this for clarity on the mess what extensions are.. but I've removed it and tried to take the mess into account in setSelectionModel(...)

The problem is that Extension interface doesn't specify other than remove() and setParent(), whereas the actual API should be about AbstractExtension.extend(...) which is used here.

Not really more I can do in this patch.
</blockquote></details>

Agreed. That's why I mentioned that `Extension` here  is weird as a superinterface.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3153 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCjTRkEIcZH7Js5WXE:-KYCjTRk4OzDrUYfiSgJ:b-4ak08e) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/main/java/com/vaadin/ui/Grid.java#L3153)):*
> ```Java
>      * @see #setSelectionModel(GridSelectionModel)
>      */
>     public GridSelectionModel<T> setSelectionMode(SelectionMode selectionMode) {
> ```

What about getters for Mode and model ?

I think :
- getter from a Mode should be just a usual (public) getter.
- getter for a model: public usual getter and (may be in addition or may be as a replacement a getter with a parameter Class<? extends SelectionModel>. So that casting to the specific type will be made out of the box via this method. The implementation : check the model type, throw an exception if it doesn't match, make a cast if it matches).

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3184 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYChin4-NhRkBUC0qqV:-KYChin45w6BCsZGyW3C:b-cocsxk) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/main/java/com/vaadin/ui/Grid.java#L3184)):*
> ```Java
>                     .setSelectAllCheckBoxVisible(selectAllCheckBoxVisible);
>         } else {
>             throw new IllegalStateException(
> ```

This is the reason why the `setSelectAllCheckBoxVisible` method is in the wrong place.
It's weird to have a method in the `Grid` if it can be used only for Multi selection model AND ONLY if this model has one specific implementation.
I would consider to remove it from there.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3200 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCiaGI1qvCLopZKEMi:-KYCiaGI4WAuean_V3UY:b-pck0ge) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/main/java/com/vaadin/ui/Grid.java#L3200)):*
> SelectAllCheckBoxVisible
By the way (my fault I guess from the other patch): the name of the enumeration is confusing.
It's rather a statement which actually excludes everything except VISIBLE.
I would say it should be `SelectAllVisiblity`.

And here again: remove this method at all ?

---

*[server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java, line 565 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCkdbC9ME6gIkkJfVD:-KYCkdbCDtMj2Kt5m6Z3:b-63wadh) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java#L565)):*
> ```Java
>         };
> 
>         Grid<String> grid = new Grid<String>() {
> ```

`CustomSelectionModelGrid` could be designed so that it accepts any model in the CTOR.
Then current `CustomSelectionModelGrid` class can be just a child of this model with a custom model impl.
Here the generic one can be used.

But minor.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 2 of 5 files at r2.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

pleku commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java, line  at r1](https://reviewable.io:443/reviews/vaadin/framework/83#-KXzceQd1OuwL8m7fwfv:-KYCnKQm-O6nxw8MWhz4:b-5c0am9) ([raw file](https://github.com/vaadin/framework/blob/79d6f7c773225168baac376285fff471760e3207/compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java#L)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

OK.......
But this is not a UI test (so it's executed). Smoke tests are good but is it really necessary to delete existing tests ? (which are already written)
Well, up to you.
</blockquote></details>

Ok, added back. I guess it makes sense for server side tests. UI tests are a bit different though.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3153 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCjTRkEIcZH7Js5WXE:-KYCnvYGDYGVFHewC1a8:b7sthud) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/main/java/com/vaadin/ui/Grid.java#L3153)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

What about getters for Mode and model ?

I think :
- getter from a Mode should be just a usual (public) getter.
- getter for a model: public usual getter and (may be in addition or may be as a replacement a getter with a parameter Class<? extends SelectionModel>. So that casting to the specific type will be made out of the box via this method. The implementation : check the model type, throw an exception if it doesn't match, make a cast if it matches).
</blockquote></details>

• I'm uncertain about getter for SelectionMode, since if someone uses a custom selection model, not an inbuilt model, it won't map the correct things.

• There is already a `getSelectionModel: GridSelectionModel<T>`

• Using a getter of signature
`<SELECTIONMODEL extends GridSelectionModel<T>> SELECTIONMODEL getSelectionModel(Class<SELECTIONMODEL> type)`will result into unchecked operation in the calling code because of the type parameter in the selection model, so don't think it is any use compared to getSelectionModel() and casting that to your correct type. If I'm doing something wrong here, lets look at it in a separate patch ?

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3184 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYChin4-NhRkBUC0qqV:-KYCvgjm1vY-CBrFuhXU:b-8oltga) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/main/java/com/vaadin/ui/Grid.java#L3184)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

This is the reason why the `setSelectAllCheckBoxVisible` method is in the wrong place.
It's weird to have a method in the `Grid` if it can be used only for Multi selection model AND ONLY if this model has one specific implementation.
I would consider to remove it from there.
</blockquote></details>

Done.

Looking at the API now, I think we  still might want to have MultiSelectGrid and SingleSelectGrid inheriting Grid with direct  access to the unchangeable selection model, but that is another question. I've added a ticket about selection model API review already.

---

*[server/src/main/java/com/vaadin/ui/Grid.java, line 3200 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCiaGI1qvCLopZKEMi:-KYCwQOI4o1CmOIC7Mzc:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/main/java/com/vaadin/ui/Grid.java#L3200)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

> SelectAllCheckBoxVisible
By the way (my fault I guess from the other patch): the name of the enumeration is confusing.
It's rather a statement which actually excludes everything except VISIBLE.
I would say it should be `SelectAllVisiblity`.

And here again: remove this method at all ?
</blockquote></details>

Done.

---

*[server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java, line 565 at r2](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCkdbC9ME6gIkkJfVD:-KYCxIJ53wBtGJZgwnYL:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/e2056af8ddb01ea0c7922e1e9dc631a9541ad345/server/src/test/java/com/vaadin/tests/components/grid/GridMultiSelectionModelTest.java#L565)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

`CustomSelectionModelGrid` could be designed so that it accepts any model in the CTOR.
Then current `CustomSelectionModelGrid` class can be just a child of this model with a custom model impl.
Here the generic one can be used.

But minor.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java, line 377 at r3](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCyk4c-SY6pKO8IAga:-KYCyk4c-y4lIYYs7BEf:b-d0wxhj) ([raw file](https://github.com/vaadin/framework/blob/b11a5e894486e6a3b2a5b88b298f10944072459a/compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java#L377)):*
> ```Java
>                 grid.getSelectedRow());
>     }
> }
> ```

new line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

pleku commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java, line 377 at r3](https://reviewable.io:443/reviews/vaadin/framework/83#-KYCyk4c-SY6pKO8IAga:-KYDW_Wi8bFukG5R_LuO:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b11a5e894486e6a3b2a5b88b298f10944072459a/compatibility-server/src/test/java/com/vaadin/v7/tests/server/component/grid/GridSelectionTest.java#L377)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

new line
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 1 of 2 files at r4.
Review status: 16 of 17 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->

denis-anisimov commented:
Reviewed 1 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/83)*
<!-- Sent from Reviewable.io -->
